controller/work_item_labels_blackbox_test.go

```diff
@@ -0,0 +1,227 @@
+package controller_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-wit/account"
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/app/test"
+	. "github.com/fabric8-services/fabric8-wit/controller"
+	"github.com/fabric8-services/fabric8-wit/gormapplication"
+	"github.com/fabric8-services/fabric8-wit/gormsupport/cleaner"
+	"github.com/fabric8-services/fabric8-wit/gormtestsupport"
+	"github.com/fabric8-services/fabric8-wit/label"
+	"github.com/fabric8-services/fabric8-wit/resource"
+	testsupport "github.com/fabric8-services/fabric8-wit/test"
+	tf "github.com/fabric8-services/fabric8-wit/test/testfixture"
+	wittoken "github.com/fabric8-services/fabric8-wit/token"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestWorkItemLabelREST struct {
+	gormtestsupport.DBTestSuite
+	db           *gormapplication.GormDB
+	ctx          context.Context
+	clean        func()
+	testIdentity account.Identity
+}
+
+func TestRunWorkItemLabelREST(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &TestWorkItemLabelREST{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
+}
+
+func (l *TestWorkItemLabelREST) SetupTest() {
+	l.db = gormapplication.NewGormDB(l.DB)
+	l.clean = cleaner.DeleteCreatedEntities(l.DB)
+	req := &http.Request{Host: "localhost"}
+	params := url.Values{}
+	l.ctx = goa.NewContext(context.Background(), nil, req, params)
+	fixture := tf.NewTestFixture(l.T(), l.DB, tf.Identities(1))
+	require.NotNil(l.T(), fixture)
+	require.Len(l.T(), fixture.Identities, 1)
+	l.testIdentity = *fixture.Identities[0]
+}
+
+func (l *TestWorkItemLabelREST) TearDownTest() {
+	l.clean()
+}
+
+func (l *TestWorkItemLabelREST) SecuredController() (*goa.Service, app.WorkitemController) {
+	pub, _ := wittoken.RSAPublicKey()
+	svc := testsupport.ServiceAsUser("WorkItemLabel-Service", wittoken.NewManager(pub), l.testIdentity)
+	return svc, NewWorkitemController(svc, l.db, l.Configuration)
+}
+
+func (l *TestWorkItemLabelREST) UnSecuredController() (*goa.Service, app.WorkitemController) {
+	svc := goa.New("WorkItemLabel-Service")
+	return svc, NewWorkitemController(svc, l.db, l.Configuration)
+}
+
+func (l *TestWorkItemLabelREST) TestAttachDetachLabelToWI() {
```

Comment:
is there any way to make this test smaller ? if you're testing multiple cases, we'd rather have multiple tests, instead.

Reply:
possible but it is much of redundancy... I will try to break few.

Reply:
@xcoulon I added few tests please see  https://github.com/fabric8-services/fabric8-wit/pull/1627/commits/20a09410a4cb30b338229bbad7a31ddf4431f377